src/librustc_resolve/macros.rs

```diff
@@ -439,25 +439,32 @@ impl<'a> Resolver<'a> {
                 return Err(Determinacy::Determined);
             }
 
-            let def = match self.resolve_path(&path, Some(MacroNS), false, span, None) {
-                PathResult::NonModule(path_res) => match path_res.base_def() {
-                    Def::Err => Err(Determinacy::Determined),
-                    def @ _ => {
-                        if path_res.unresolved_segments() > 0 {
-                            self.found_unresolved_macro = true;
-                            self.session.span_err(span, "fail to resolve non-ident macro path");
-                            Err(Determinacy::Determined)
-                        } else {
-                            Ok(def)
+            let def = {
+                let resolve_path = ResolvePath {
```

Comment:
this should be constructed immediately after we construct `path`

Reply:
The problem I had with this is, written like this, we need the borrow to end on the `Vec` so we can do `push((path.into_boxed_slice(), span))` later on in the function. I couldn't find a way to turn a `&[T]` directly into a `Box<[T]>`, is there a way to do that?